### PR TITLE
Testing fixes

### PR DESCRIPTION
**Pull request by ssciolist**
# Changes Proposed:
- adds an error exception for game not found to game controller
-  adds api keys to shots spec 
-
# Fixes:
- modifies Game factory to reflect what is currently needed
-

# Checklist:
* Did you add to application.yml? NO
* Passing RSPEC? One failure in the shots spec
* Passing Spec Harness? until line 73
* Ran on Local Host?

# Mentions:

### CODE REVIEW

File: app/controllers/api/v1/games_controller.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module Api
   module V1
     class GamesController < ActionController::API
+      rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
```
Comment (lilwillifo) [inline app/controllers/api/v1/games_controller.rb]: fancy! I like it.

File: spec/requests/shots_spec.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@
     let(:player_1)   { Player.new(Board.new(4),'kasjdfhlai4oeiufrwh')}
```
Comment (lilwillifo) [inline spec/requests/shots_spec.rb]: I think we could set these up in the game factory bot rather than in this test file. small refactor!

File: app/controllers/api/v1/games_controller.rb (hunk 1 above)
Comment (ssciolist) [inline app/controllers/api/v1/games_controller.rb]: @icorson3, @jmejia, I tried this method in both API controller and application controller and neither worked. File structure looks like this
screen shot 2018-05-17 at 7 09 40 pm ... any guidance on what's wrong?

File: .ruby-version [hunk 3]
```diff
@@ -1 +1 @@
-system
+2.4.1
```
Comment (ssciolist) [inline .ruby-version]: @jmejia @icorson3 Is this file needed? We generated in the course of today's work (so maybe this was prompted by version changes from the am lesson setup?) but it has mostly caused issues, namely with rspec running.

File: app/controllers/api/v1/games_controller.rb (hunk 1 above)
Comment (jmejia) [reply app/controllers/api/v1/games_controller.rb]: @ssciolist I've never used `rescue_from`.  What's your goal here? Also not sure what it means for it to not work. Can you clarify?
